### PR TITLE
add a way to sort printable reports

### PR DESCRIPTION
**Pull request by ricardofreitasrocha**

### CODE REVIEW

File: server/routes/reporting/standard-report/definitions.js [hunk 1]
```diff
@@ -1418,6 +1456,7 @@
           'courtName',
           'totalExcusedAndDisqualified',
         ],
+        defaultSortColumn: 'lastName',
         grouped: {
           groupHeader: true,
         },
```
Comment (callum-r-young) [inline server/routes/reporting/standard-report/definitions.js:1463]: add `totals: true`

report missing totals from figma

**Review by callum-r-young — CHANGES_REQUESTED** [review]

File: server/routes/reporting/standard-report/standard-report-print.js [hunk 2]
```diff
@@ -292,6 +294,56 @@ async function standardReportPrint(app, req, res, reportKey, data) {
...
+      return _b.localeCompare(_a);
+    } else {
+      return _a.localeCompare(_b);
+    }
+  }
+}
+
+function formatSortableData(a, b, sortBy) {
+  let _a = a[snakeToCamel(sortBy)];
+  let _b = b[snakeToCamel(sortBy)];
```
[the remark this answers is not in the document]
Comment (ricardofreitasrocha) [reply server/routes/reporting/standard-report/standard-report-print.js:339]: yes 😄 Thank you

[the remark this answers is not in the document]
Comment (ricardofreitasrocha) [reply server/routes/reporting/standard-report/standard-report-print.js]: updated

File: server/routes/reporting/standard-report/definitions.js (hunk 1 above)
Comment (ricardofreitasrocha) [reply server/routes/reporting/standard-report/definitions.js:1464]: updated

File: server/routes/pool-management/create-pool/summon-citizens/summon-citizens.controller.js [hunk 3]
```diff
@@ -315,7 +315,7 @@
         , errorUrl;
 
       if (page === 'summon-citizens') {
-        currentBureauDeferrals = req.session[`summonJurors-${poolNumber}`].availableBureauDeferrals;
+        currentBureauDeferrals = req.session[`summonJurors-${poolNumber}`].currentBureauDeferrals;
```
[the remark this answers is not in the document]
Comment (ricardofreitasrocha) [reply server/routes/pool-management/create-pool/summon-citizens/summon-citizens.controller.js]: this change was left from a rebase.

File: server/routes/reporting/standard-report/definitions.js [hunk 4]
```diff
@@ -1119,6 +1151,7 @@
           'courtName',
           'totalOnCall',
         ],
+        defaultSortColumn: 'lastName',
       },
       'trial-attendance': {
```
Comment (callum-r-young) [inline server/routes/reporting/standard-report/definitions.js:1156]: BespokeReport.insertRows function seems to be breaking the sorting on the report while viewing, maybe make this unsortable or we need to stop these inserted rows from being sorted as they should always be at bottom of each table

File: server/routes/reporting/standard-report/standard-report-print.js [hunk 5]
```diff
@@ -292,6 +304,70 @@ async function standardReportPrint(app, req, res, reportKey, data) {
   }
 };
 
+function sortTableData({ sortBy, sortDirection }, tableData, reportData) {
+  if (reportData.bespokeReport) return; // we do not sort bespoke reports for now
```
Comment (callum-r-young) [inline server/routes/reporting/standard-report/standard-report-print.js:308]: Maybe change this to;

`if (reportData.bespokeReport.body) return;`

I have tested each of the reports that are bespoke but dont contain a seperate body file, and these all sort and work as expected (other than trial-attendance, which is in another comment). 
Would mean reports such as persons attending can be sorted however they please for scanning and checking in jurors

**Review by callum-r-young — CHANGES_REQUESTED** [review]

File: server/routes/reporting/standard-report/definitions.js (hunk 4 above)
Comment (ricardofreitasrocha) [reply server/routes/reporting/standard-report/definitions.js:1156]: I missed one 😄 yes, we need to find a good way of injecting an attribute of `fixed-row-index` to block it from being sorted... or I can try and do something clever with the sorter.

File: server/routes/reporting/standard-report/standard-report-print.js [hunk 6]
```diff
@@ -305,7 +305,7 @@ async function standardReportPrint(app, req, res, reportKey, data) {
 };
 
 function sortTableData({ sortBy, sortDirection }, tableData, reportData) {
-  if (reportData.bespokeReport) return; // we do not sort bespoke reports for now
+  if (reportData.bespokeReport.body) return; // we do not sort bespoke reports for now
```
Comment (callum-r-young) [inline server/routes/reporting/standard-report/standard-report-print.js:308]: CHange to the following?
`if (reportData.bespokeReport?.body) return;`

Breaks when report definitions dont have bespokeReport

**Review by callum-r-young — CHANGES_REQUESTED** [review]

**Review by callum-r-young — APPROVED** [review]

**Pull request merged**